data/src/main/java/com/microsoft/azure/kusto/data/auth/KcsbKeywords.java

```diff
@@ -0,0 +1,81 @@
+package com.microsoft.azure.kusto.data.auth;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.kusto.data.Utils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+public class KcsbKeywords {
+    public KeywordData[] keywords;
+
+    private final Map<String, KnownKeywords> lookup = new HashMap<>();
+
+    private static KcsbKeywords instance = null;
+    private static final Object object = new Object();
+
+    public static KcsbKeywords getInstance() {
+        if (instance != null)
+            return instance;
+        synchronized (object) {
+            if (instance == null) {
+                instance = readInstance();
+            }
+        }
+        return instance;
+    }
+
+    // For Deserialization
+    public KcsbKeywords() {
+    }
+
+    public static String normalize(String keyword) {
+        return keyword.toLowerCase().replace(" ", "");
+    }
+
+    @NotNull
+    private static KcsbKeywords readInstance() {
+        try {
+            ObjectMapper objectMapper = Utils.getObjectMapper();
+            try (InputStream resourceAsStream = KcsbKeywords.class.getResourceAsStream(
+                    "/kcsb.json")) {
+                KcsbKeywords value = objectMapper.readValue(resourceAsStream, KcsbKeywords.class);
+
+                // Validate the keywords
+                for (KeywordData keywordData : value.keywords) {
+                    KnownKeywords keyword = KnownKeywords.knownKeywords.get(keywordData.name);
```

Comment:
Is `get()` fails it means there's a keyword in the json we did not code.
Will there be an explicit error on this?